### PR TITLE
Explain how to track tool dependencies

### LINKED ISSUE #92: docs: mention the tools.go approach to pin tool version
Hello, thanks for pegomock! :-)

would it make sense to mention the tools.go approach to ensure that the pegomock executable is the same version for everybody using it an given project? I learned this from [counterfeiter documentation](https://github.com/maxbrunsfeld/counterfeiter#step-1---create-toolsgo) and is explained in the [Go module docs](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module).

If you agree, I would be happy to provide a PR. Where in the README would you like it? I was thinking just after section "Generating Your First Mock and Using It".

### PR DESCRIPTION
**Pull request by marco-m**
@petergtz this closes #92 as discussed. On the other hand, while writing the documentation, I realized that it is more involved than what I was thinking (as you can see in the PR, it makes really sense only if ones has a build script).

So now I am not so sure anymore of the value of mentioning this technique. I am OK if at the end you decide to reject.

TODO

- [ ] Validate problem with Go 1.13 after merge to master of develop, see comments

Closes #92

### CODE REVIEW

**Comment by petergtz** [comment]
@marco-m Thanks a lot for your work. I need some time to fully digest your change and comment. I think in general it makes sense to make people aware of the best practice, because it's a bit hidden in the big modules wiki and we had this [question how to keep things in sync](https://github.com/petergtz/pegomock/issues/74#issuecomment-455331984) already here. I simply haven't used this mechanism myself yet, so I need to familiarize myself more with it first.

Will get back to you.

File: README.md [hunk 1]
```diff
@@ -116,6 +116,33 @@ display.Show("Hello World!")
...
+// +build tools
+
+// This file will never be compiled due (see the build constraint); it is used
+// simply to record dependencies on build tools.
+// See https://github.com/go-modules-by-example/index/blob/master/010_tools/README.md
+
+package tools
+
+import (
+	_ "github.com/petergtz/pegomock/pegomock"
```
Comment (jpopadak) [inline README.md]: One thing to note, this does not work with 1.13.

```
go get -tags tools ./internal/...
internal/tools.go:6:2: import "github.com/petergtz/pegomock/pegomock" is a program, not an importable package
```

File: README.md [hunk 2]
```diff
@@ -116,6 +116,33 @@ display.Show("Hello World!")
...
+)
+```
+2. Set `$GOBIN` to a `bin` directory relative to your repo (this defines where tool dependencies will be installed)
+2. Install the tool with `go install`
+3. Use that path in your build script.
```
Comment (petergtz) [inline README.md]: How about:
> 3. Use `$GOBIN/pegomock` to generate your mocks. 

That would avoid talking about a build script that you worried about.

File: README.md [hunk 3]
```diff
@@ -116,6 +116,33 @@ display.Show("Hello World!")
 display.VerifyWasCalledOnce().Show("Hello World!")
 ```
 
+Tracking the pegomock tool in your project
```
Comment (petergtz) [inline README.md]: Thinking about where to best insert this section, how about adding this section after https://github.com/petergtz/pegomock#installation? Would feel very natural to me to look for something like this here.

File: README.md (hunk 1 above)
Comment (petergtz) [reply README.md]: @jpopadak Could you elaborate? I just tried this and it works for me. Here's what I've done:
```shell
$ mkdir -p /home/peter/tmp/blump
$ cd /home/peter/tmp/blump
$ git init -q
$ git remote add origin https://github.com/petergtz/blump
$ go mod init
go: creating new go.mod: module github.com/petergtz/blump
$ export GOBIN=$PWD/bin
$ export PATH=$GOBIN:$PATH

$ cat tools.go
// +build tools

package tools

import (
	_ "github.com/petergtz/pegomock/pegomock"
)

$ cat go.mod
module github.com/petergtz/blump

go 1.13

require (
	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
	github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d // indirect
	github.com/petergtz/pegomock v2.6.0+incompatible // indirect
	golang.org/x/tools v0.0.0-20190927191325-030b2cf1153e // indirect
	gopkg.in/alecthomas/kingpin.v2 v2.2.6 // indirect
)

$ cat main.go
package main

import "fmt"

func main() {
	fmt.Println("Hello")
}

$ go install github.com/petergtz/pegomock/pegomock
go: finding github.com/petergtz/pegomock v2.6.0+incompatible
go: downloading github.com/petergtz/pegomock v2.6.0+incompatible
go: extracting github.com/petergtz/pegomock v2.6.0+incompatible
go: finding gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: finding golang.org/x/tools latest
go: downloading gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: downloading golang.org/x/tools v0.0.0-20190927191325-030b2cf1153e
go: extracting gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: extracting golang.org/x/tools v0.0.0-20190927191325-030b2cf1153e
go: finding github.com/alecthomas/template latest
go: finding github.com/alecthomas/units latest
go: downloading github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
go: downloading github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d
go: extracting github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
go: extracting github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d

$ which pegomock 
/home/peter/tmp/blump/bin/pegomock
```

**Review by petergtz — CHANGES_REQUESTED** [review]
@marco-m It looks pretty good to me. Just 2 minor comments. After those, and after clarifying with @jpopadak I will merge it.

File: README.md (hunk 2 above)
Comment (marco-m) [reply README.md]: Done. I made it even more explicit, please have a look.

File: README.md (hunk 3 above)
Comment (marco-m) [reply README.md]: done

**Comment by marco-m** [comment]
@petergtz did the changes are requested. I committed each change separately to ease review; on the other hand I suggest to perform a squash merge, it doesn't make sense to keep the individual commits in this PR.

**Review by petergtz — APPROVED** [review]

**Pull request merged**